test/tools/unified-spec-runner/unified-utils.ts

```diff
@@ -146,12 +146,16 @@ export async function topologySatisfies(
 }
 
 export async function isAnyRequirementSatisfied(ctx, requirements, client) {
+  const skipTarget = ctx.currentTest || ctx.test;
+  const skipReasons = [];
   for (const requirement of requirements) {
     const met = await topologySatisfies(ctx, requirement, client);
     if (met) {
       return true;
     }
+    skipReasons.push(skipTarget.skipReason);
   }
+  skipTarget.skipReason = skipReasons.join(' OR ');
```

Comment:
[Example log line for the curious](https://parsley.mongodb.com/evergreen/mongo_node_driver_next_rhel80_large_Node20_test_3.6_replica_set_patch_ecfc6157786360832d1afb4294d76f83a90a9d70_65ce3fc03e8e863142e5fd7b_24_02_15_16_45_53/0/task?bookmarks=0,8949&shareLine=4109)